### PR TITLE
Fix the error message in the _check_dtype_and_dim function

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Fixes #

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by willschlitzer** [comment]
Is it incorrect to refer to the "dimensions" plural for an array (assuming it is more than 1d)?

**Comment by seisman** [comment]
> Is it incorrect to refer to the "dimensions" plural for an array (assuming it is more than 1d)?

I'm not sure. Perhaps "dimensions" is more general and better.

File: pygmt/clib/session.py [hunk 1]
```diff
@@ -677,12 +677,12 @@ def _parse_constant(self, constant, valid, valid_modifiers=None):
 
     def _check_dtype_and_dim(self, array, ndim):
         """
-        Check that a numpy array has the given dimensions and is a valid data
+        Check that a numpy array has the given dimension and is a valid data
```
Comment (seisman) [inline pygmt/clib/session.py:680]:
```suggestion
        Check that a numpy array has the given dimensions and is a valid data
```

File: pygmt/clib/session.py [hunk 2]
```diff
@@ -694,8 +694,8 @@ def _check_dtype_and_dim(self, array, ndim):
 
         Raises
         ------
-        GMTCLibError
-            If the array has the wrong dimensions or is an unsupported data
+        GMTInvalidInput
+            If the array has the wrong dimension or is an unsupported data
```
Comment (seisman) [inline pygmt/clib/session.py:698]:
```suggestion
            If the array has the wrong dimensions or is an unsupported data
```

**Review by willschlitzer — APPROVED** [review]

File: pygmt/clib/session.py [hunk 3]
```diff
@@ -717,7 +717,7 @@ def _check_dtype_and_dim(self, array, ndim):
         """
         # check the array has the given dimension
         if array.ndim != ndim:
-            raise GMTInvalidInput(f"Expected a numpy 1d array, got {array.ndim}d.")
+            raise GMTInvalidInput(f"Expected a numpy {ndim}d array, got {array.ndim}d.")
```
Comment (yvonnefroehlich) [inline pygmt/clib/session.py:720]: For consistency with PR #2252, for the rest please see PR #2307.
```suggestion
            raise GMTInvalidInput(f"Expected a numpy {ndim}-D array, got {array.ndim}-D.")
```

**Review by yvonnefroehlich — COMMENTED** [review]
The change regarding the consistent usage of x-D looks good!

Regarding whether it is the dimension (singular) or the dimensions (plural) of an array, I am not sure 🙁. Currently it is not consistent. For _me_ for all four occurrences in `_check_dtype_and_dim` singular _sounds_ better, as `ndim` is  one single integer value indicating the overall number of dimensions:

- lines 680-681:
```python
        Check that a numpy array has the given dimensions and is a valid data
        type.   # -> change to dimension (singular)
```

- lines 687-688:
```python
        ndim : int
            The desired dimension of the array.
```

- lines 697-699:
```python
        GMTCLibError
            If the array has the wrong dimensions or is an unsupported data
            type.  # -> change to dimension (singular)
```

- lines 718-720:
```python
        # check the array has the given dimension
        if array.ndim != ndim:
            raise GMTInvalidInput(f"Expected a numpy 1-D array, got {array.ndim}-D.")
```

Plural is used in:
`create_data`
- lines 529-533:
```python
        dim : list of 4 integers
            The dimensions of the dataset. See the documentation for the GMT C
            API function ``GMT_Create_Data`` (``src/gmt_api.c``) for the full
            range of options regarding 'dim'. If ``None``, will pass in the
            NULL pointer.
``` 
`extract_region`
- lines 1489-1491:
```python
        * wesn : 1-D array   # <- is the * needed or a typo?
            A numpy 1-D array with the west, east, south, and north dimensions
            of the current figure.  # <- maybe limits is a better word here?
```

**Comment by seisman** [comment]
> Regarding whether it is the dimension (singular) or the dimensions (plural) of an array, I am not sure 🙁. Currently it is not consistent. For _me_ for all four occurrences in `_check_dtype_and_dim` singular _sounds_ better, as `ndim` is one single integer value indicating the overall number of dimensions:

The numpy documentation for [`np.ndarray.ndim`](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.ndim.html) says:

> Number of array dimensions.

So I think we can follow their style.

File: pygmt/clib/session.py [hunk 4]
```diff
@@ -717,7 +717,9 @@ def _check_dtype_and_dim(self, array, ndim):
         """
         # check the array has the given dimension
```
Comment (michaelgrund) [inline pygmt/clib/session.py:718]:
```suggestion
        # check that the array has the given dimension
```

File: pygmt/clib/session.py [hunk 5]
```diff
@@ -717,7 +717,9 @@ def _check_dtype_and_dim(self, array, ndim):
...
         if array.ndim != ndim:
-            raise GMTInvalidInput(f"Expected a numpy 1d array, got {array.ndim}d.")
+            raise GMTInvalidInput(
+                f"Expected a numpy {ndim}-D array, got {array.ndim}-D."
+            )
 
         # check the array has a valid/known data type
```
Comment (michaelgrund) [inline pygmt/clib/session.py:724]:
```suggestion
        # check that the array has a valid/known data type
```

**Review by michaelgrund — APPROVED** [review]

File: pygmt/clib/session.py [hunk 6]
```diff
@@ -682,7 +682,7 @@ def _check_dtype_and_dim(self, array, ndim):
 
         Parameters
         ----------
-        array : numpy array
+        array : numpy.ndarray
             The array to be tested.
         ndim : int
             The desired dimension of the array.
```
Comment (yvonnefroehlich) [inline pygmt/clib/session.py:688]: This has to be changed similar above in the lines 680-681, please. (GitHub does not allow me to make a suggestion there).
```suggestion
        ndim : int
            The desired number of dimensions of the array.
```

File: pygmt/clib/session.py [hunk 7]
```diff
@@ -694,7 +694,7 @@ def _check_dtype_and_dim(self, array, ndim):
 
         Raises
         ------
-        GMTCLibError
+        GMTInvalidInput
             If the array has the wrong dimensions or is an unsupported data
             type.
```
Comment (yvonnefroehlich) [inline pygmt/clib/session.py:699]:
```suggestion
            If the array has the wrong number of dimensions or
            is an unsupported data type.
```

File: pygmt/clib/session.py (hunk 4 above)
Comment (yvonnefroehlich) [inline pygmt/clib/session.py:718]: Upper-case beginning for consistency, together with the suggestion by @michaelgrund.
```suggestion
        # Check that the array has the given number of dimensions.
```

File: pygmt/clib/session.py (hunk 5 above)
Comment (yvonnefroehlich) [inline pygmt/clib/session.py:724]: Upper-case beginning for consistency, together with the suggestion by @michaelgrund.
```suggestion
        # Check that the array has a valid/known data type
```

**Review by michaelgrund — APPROVED** [review]

**Review by yvonnefroehlich — APPROVED** [review]

**Pull request merged**